### PR TITLE
feat: Multi region support

### PR DESCRIPTION
**Pull request by dskart**
## What

- allows aws requests to work across different regions
- adds unit and integration tests for multi-region support

### CODE REVIEW

File: boltrouter/main_test.go [hunk 1]
```diff
@@ -53,3 +57,55 @@ func SetupQuickSilverMock(t *testing.T, ctx context.Context, logger *zap.Logger)
...
+		config.WithRegion(region),
+	)
+	require.NoError(t, err)
+
+	// ret.S3Client = s3.NewFromConfig(cfg, func(o *s3.Options) {
+	// 	if requestStyle == pathStyle {
+	// 		o.UsePathStyle = true
+	// 	}
+	// 	o.Region = region
+	// })
```
Comment (kmushegi) [inline boltrouter/main_test.go:100]: need this still?

File: boltrouter/source_bucket.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+		Bucket: "n-auth-dummy",
+		Style:  nAuthDummy,
+	}
+
+	isVirtualHostedStyle := false
+	split := strings.Split(req.Host, ".")
+	if len(split) > 1 {
+		if _, err := strconv.Atoi(split[0]); err != nil {
+			// is not a number, so it is a bucket name
+			isVirtualHostedStyle = true
```
Comment (kmushegi) [inline boltrouter/source_bucket.go:46]: pathstyle host looks like this `Host: s3.us-east-1.amazonaws.com`
virtual hosted host looks like this `Host: DOC-EXAMPLE-BUCKET1.eu.s3.eu-west-1.amazonaws.com`

so split[0] yields `s3` and `DOC-EXAMPLE-BUCKET1` respectively?

**Review by kmushegi — COMMENTED** [review]
looks good overall and tested w aws cli on linux

File: boltrouter/main_test.go (hunk 1 above)
Comment (dskart) [reply boltrouter/main_test.go]: good catch, fixed

**Pull request merged**